proof-producer/libs/assigner/include/nil/proof-generator/assigner/trace_parser.hpp

```diff
@@ -137,11 +141,46 @@ namespace nil {
             return res;
         }
 
-        [[nodiscard]] std::optional<std::unordered_map<std::string, std::string>> deserialize_bytecodes_from_file(const boost::filesystem::path& bytecode_trace_path) {
+        [[nodiscard]] std::string get_trace_extension(const boost::filesystem::path& trace_path) {
+            return trace_path.extension().string();
+        }
+
+        using TraceIndex = uint64_t; // value expected to be the same for all traces from the same set 
+        class trace_index_mismatch: public std::exception {        
+        public:
+            [[nodiscard]] const char* what() const noexcept override {
+                return "Invalid trace index";
+            }
+        };
+
+        inline void check_trace_index(TraceIndex base, TraceIndex index) {
+            if (base != TraceIndex{} && index != base) {
+                throw trace_index_mismatch();
```

Comment:
why do you want to  introduce exceptions here?
may be just return `nullopt` from deserialization function?

Reply:
initially thought to handle the error in one place but at the middle of the way decided optionally ignore index mismatch. Rewrote with `std::nullopt` and warning log